src/hotspot/share/runtime/os.cpp

```diff
@@ -1480,7 +1480,7 @@ size_t os::page_size_for_region(size_t region_size, size_t min_pages, bool must_
 
     for (size_t page_size = page_sizes().largest(); page_size != 0;
          page_size = page_sizes().next_smaller(page_size)) {
-      if (page_size <= max_page_size) {
+      if (page_size <= max_page_size && page_size > (size_t)vm_page_size()) {
```

Comment:
Is this really needed?

Reply:
Probably not. I am doublechecking with some testing.